pkg/ragengine/controllers/ragengine_controller.go

```diff
@@ -152,8 +153,34 @@ func (c *RAGEngineReconciler) applyRAG(ctx context.Context, ragEngineObj *kaitov
 
 		if err = resources.GetResource(ctx, ragEngineObj.Name, ragEngineObj.Namespace, c.Client, deployment); err == nil {
 			klog.InfoS("An inference workload already exists for ragengine", "ragengine", klog.KObj(ragEngineObj))
-			return
+			if deployment.Annotations[kaitov1alpha1.RAGEngineRevisionAnnotation] != revisionStr {
+				var volumes []corev1.Volume
+				var volumeMounts []corev1.VolumeMount
+				shmVolume, shmVolumeMount := utils.ConfigSHMVolume(*ragEngineObj.Spec.Compute.Count)
```

Comment:
Do we need shmvolumes for rag engine?

Reply:
I believe only if they use GPU node, and then run the embedding model locally on that